src/main/java/com/configcat/EvaluationDetails.java

```diff
@@ -0,0 +1,107 @@
+package com.configcat;
+
+/**
+ * Additional information about flag evaluation.
+ */
+public class EvaluationDetails<T> {
+    private final T value;
+    private final String key;
+    private final String variationId;
+    private final User user;
+    private final boolean isDefaultValue;
+    private final String error;
+    private final long fetchTimeUnixMilliseconds;
+    private final RolloutRule matchedEvaluationRule;
+    private final PercentageRule matchedEvaluationPercentageRule;
+
+    public EvaluationDetails(T value,
+                             String key,
+                             String variationId,
+                             User user,
+                             boolean isDefaultValue,
+                             String error,
+                             long fetchTimeUnixMilliseconds,
+                             RolloutRule matchedEvaluationRule,
+                             PercentageRule matchedEvaluationPercentageRule) {
+        this.value = value;
+        this.key = key;
+        this.variationId = variationId;
+        this.user = user;
+        this.isDefaultValue = isDefaultValue;
+        this.error = error;
+        this.fetchTimeUnixMilliseconds = fetchTimeUnixMilliseconds;
+        this.matchedEvaluationRule = matchedEvaluationRule;
+        this.matchedEvaluationPercentageRule = matchedEvaluationPercentageRule;
+    }
+
+    static <T> EvaluationDetails<T> fromError(String key, T defaultValue, String error, User user) {
+        return new EvaluationDetails<>(defaultValue, key, "", user, true, error, Constants.DISTANT_PAST, null, null);
+    }
+
+    <TR> EvaluationDetails<TR> asTypeSpecific() {
+        return new EvaluationDetails<>((TR) value, key, variationId, user, isDefaultValue, error, fetchTimeUnixMilliseconds, matchedEvaluationRule, matchedEvaluationPercentageRule);
+    }
+
+    /**
+     * The evaluated value of the feature flag or setting.
+     */
+    public T getValue() {
+        return value;
+    }
+
+    /**
+     * The key of the evaluated feature flag or setting.
+     */
+    public String getKey() {
+        return key;
+    }
+
+    /**
+     * The evaluated variation ID (analytics).
```

Comment:
is the variation id evaluated?

Reply:
This still needs to be discussed.

Reply:
The variation ID comes form the RolloutEvaluator, true, nothing really evaluated, It comes from a Rollout or from a Percentage rule. Still, I think it is fine like that, the DTO don't have to know how the evaluation happens. So it can say it is evaluated.

Reply:
Discussed over slack, adding it here for the record: `VariationID: The ID of the evaluated value. Useful for analytics.`